libwallet/dex_wallets_loader.go

```diff
@@ -136,69 +137,71 @@ func prepareDexSupportForBTCCloneWallets() {
+	return func(settings map[string]string, chainParams *btccfg.Params) (dexbtc.CustomWallet, error) {
+		walletIDStr := settings[dexc.WalletIDConfigKey]
+		walletID, err := strconv.Atoi(walletIDStr)
+		if err != nil || walletID < 0 {
+			return nil, fmt.Errorf("invalid wallet ID %q in settings", walletIDStr)
+		}
 
-	wallet := loadWalletForDEX(walletID)
-	if wallet == nil {
-		return nil, fmt.Errorf("no wallet exists with ID %q", walletIDStr)
-	}
+		wallet := loadWalletForDEX(walletID)
+		if wallet == nil {
+			return nil, fmt.Errorf("no wallet exists with ID %q", walletIDStr)
+		}
 
-	if isLtc {
-		if walletParams := wallet.Internal().LTC.ChainParams(); !strings.EqualFold(walletParams.Name, chainParams.Name) {
-			return nil, fmt.Errorf("selected wallet is for %s network, expected %s", walletParams.Name, chainParams.Name)
+		if isLtc {
+			if walletParams := wallet.Internal().LTC.ChainParams(); !strings.EqualFold(walletParams.Name, chainParams.Name) {
+				return nil, fmt.Errorf("selected wallet is for %s network, expected %s", walletParams.Name, chainParams.Name)
+			}
+		} else {
+			if walletParams := wallet.Internal().BTC.ChainParams(); walletParams.Net != chainParams.Net {
+				return nil, fmt.Errorf("selected wallet is for %s network, expected %s", walletParams.Name, chainParams.Name)
+			}
 		}
-	} else {
-		if walletParams := wallet.Internal().BTC.ChainParams(); walletParams.Net != chainParams.Net {
-			return nil, fmt.Errorf("selected wallet is for %s network, expected %s", walletParams.Name, chainParams.Name)
+
+		if wallet.IsWatchingOnlyWallet() {
+			return nil, fmt.Errorf("cannot use watch only wallet for DEX trade")
 		}
-	}
 
-	if wallet.IsWatchingOnlyWallet() {
-		return nil, fmt.Errorf("cannot use watch only wallet for DEX trade")
-	}
+		// Ensure the wallet account exists.
+		accountNumberStr := settings[dexc.WalletAccountNumberConfigKey]
+		acctNum, err := strconv.ParseInt(accountNumberStr, 10, 64)
+		if err != nil {
+			return nil, err
+		}
 
-	// Ensure the wallet account exists.
-	accountNumberStr := settings[dexc.WalletAccountNumberConfigKey]
-	acctNum, err := strconv.ParseInt(accountNumberStr, 10, 64)
-	if err != nil {
-		return nil, err
-	}
+		accountNumber := int32(acctNum)
+		if _, err = wallet.AccountName(accountNumber); err != nil {
+			return nil, fmt.Errorf("error checking selected DEX account name: %w", err)
+		}
 
-	accountNumber := int32(acctNum)
-	if _, err = wallet.AccountName(accountNumber); err != nil {
-		return nil, fmt.Errorf("error checking selected DEX account name: %w", err)
-	}
+		if isLtc {
+			return ltc.NewDEXWallet(wallet.Internal().LTC, accountNumber, wallet.(*ltc.Asset).NeutrinoClient(), chainParams, wallet), nil
+		}
 
-	if isLtc {
-		return ltc.NewDEXWallet(wallet.Internal().LTC, accountNumber, wallet.(*ltc.Asset).NeutrinoClient(), chainParams), nil
+		return btc.NewDEXWallet(wallet.Internal().BTC, accountNumber, wallet.(*btc.Asset).NeutrinoClient(), wallet), nil
```

Comment:
why does this uses internal.BTC when it is LTC?

Reply:
It doesn't. We return at line 202 for ltc.

Reply:
Oh yes, my bad